index.js

```diff
@@ -25,56 +28,87 @@ function SonyBraviaTVAccessory(log, config) {
         .on('get', this.getOn.bind(this))
         .on('set', this.setOn.bind(this));
 
+    this.updateTimer();
+}
+
+SonyBraviaTVAccessory.prototype.runTimer = function() {
+    this.getState(function(err, isOn) {
+        if (err == null && isOn != this.isOn) {
+            this.log("State changed: %s", isOn ? "on" : "off");
+            this.service.getCharacteristic(Characteristic.On).updateValue(isOn);
+            this.isOn = isOn;
+        }
+    }.bind(this));
+}
+
+SonyBraviaTVAccessory.prototype.updateTimer = function() {
+    if (this.polling) {
+        clearTimeout(this.timer);
+        setTimeout(function() {
+            this.runTimer();
+            this.updateTimer();
+        }.bind(this), this.interval * 1000);
+    }
+}
+
+SonyBraviaTVAccessory.prototype.getState = function(callback) {
+  var postData = JSON.stringify({
+      method: 'getPowerStatus',
+      params: [],
+      id: 1,
+      version: '1.0'
+  });
+
+  request.post({
+      url: "http://" + this.ipaddress + "/sony/system",
+      headers: {
+          'X-Auth-PSK': this.psk
+      },
+      form: postData
+  }, function(err, response, body) {
+
+      if (!err && response.statusCode == 200) {
+          var json = JSON.parse(body);
+          var status = json.result[0].status;
+          var isOn = status == "active";
+          callback(null, isOn); // success
+      } else {
+          if (response != null) {
+              this.log("Error getting TV status (status code %s): %s", response.statusCode, err);
+              callback(err);
+          } else {
+              callback(null, false);
+          }
+      }
+  }.bind(this));
 }
 
 SonyBraviaTVAccessory.prototype.getOn = function(callback) {
     this.log("Getting whether Sony TV is on...");
 
-    var postData = JSON.stringify({
-        method: 'getPowerStatus',
-        params: [],
-        id: 1,
-        version: '1.0'
-    });
-
-    request.post({
-        url: "http://" + this.ipaddress + "/sony/system",
-        headers: {
-            'X-Auth-PSK': this.psk
-        },
-        form: postData
-    }, function(err, response, body) {
-
-        if (!err && response.statusCode == 200) {
-            var json = JSON.parse(body);
-            var status = json.result[0].status;
-            this.log("TV status is %s", status);
-            var isOn = status == "active";
-            callback(null, isOn); // success
-        } else {
-            if (response != null) {
-                this.log("Error getting TV status (status code %s): %s", response.statusCode, err);
-                callback(err);
-            } else {
-                this.log("Error getting TV status, assuming TV is off: %s.", err);
-                callback(null, false);
-            }
-        }
-    }.bind(this));
+    if (this.polling) {
+        callback(null, this.isOn)
+    } else {
+        this.getState(function(err, isOn) {
+              this.log("State is: %s", isOn ? "on" : "off")
```

Comment:
I don't think so... The error is already logged in getState and this function forwards the result (including the error) to the framework by calling the callback. Basicly this just resembles the original behavior from before the patch in case that polling is not explicitly enabled.